interfaces/repo.go

```diff
@@ -815,37 +818,49 @@ func (r *Repository) SnapSpecification(securitySystem SecuritySystem, snapName s
 	return spec, nil
 }
 
-// BadInterfacesError is returned when some snap interfaces could not be registered.
-// Those interfaces not mentioned in the error were successfully registered.
-type BadInterfacesError struct {
-	snap   string
-	issues map[string]string // slot or plug name => message
-}
+func (r *Repository) SanitizePlugsSlots(snapInfo *snap.Info) error {
+	err := snap.Validate(snapInfo)
```

Comment:
Is there any reason to validate the snap info at this step? Also, without this the whole method returns nothing.

Reply:
Good point, moved it back to AddSnap.